phylanx/execution_tree/compile.hpp

```diff
@@ -100,7 +100,7 @@ namespace phylanx { namespace execution_tree
     ///////////////////////////////////////////////////////////////////////////
     /// Add the given variable to the compilation environment
     PHYLANX_EXPORT compiler::function define_variable(
-        std::string const& codename, std::string name,
+        std::string const& codename, compiler::primitive_name_parts name,
```

Comment:
Is this function going to modify the `name` object? If not, please pass it as a `const&`.